src/createNavigationContainer.js

```diff
@@ -160,16 +175,7 @@ export default function createNavigationContainer<T: *>(
       }
       const nav = Component.router.getStateForAction(action, state.nav);
```

Comment:
We should use the `setState(updater)` pattern here since the new state depends on the old state.